src/common/reactNativeProjectHelper.ts

```diff
@@ -13,8 +15,45 @@ export class ReactNativeProjectHelper {
         return ["0.54.0", "0.54.1", "0.54.2", "0.54.3", "0.54.4"];
     }
 
-    public static getReactNativeVersion(projectRoot: string) {
-        return new CommandExecutor(projectRoot).getReactNativeVersion();
+    public static getReactNativeVersion(projectRoot: string): Q.Promise<string> {
+        const reactNativePackageDir = path.resolve(
+            projectRoot,
+            "node_modules",
+            "react-native"
+          );
+
+        return ReactNativeProjectHelper.getReactNativePackageVersionFromNodeModules(reactNativePackageDir)
+            .catch(err => {
+                return ReactNativeProjectHelper.getReactNativeVersionFromProjectPackage(projectRoot);
+            });
+    }
+
+    public static getReactNativePackageVersionFromNodeModules(reactNativePackageDir: string): Q.Promise<string> {
+        return new Package(reactNativePackageDir).version()
+            .catch(err => {
+                const noReactNativePackageError = ErrorHelper.getInternalError(InternalErrorCode.ReactNativePackageIsNotInstalled);
+                throw noReactNativePackageError;
+            });
+    }
+
+    public static getReactNativeVersionFromProjectPackage(cwd: string): Q.Promise<string> {
+        const rootProjectPackageJson = new Package(cwd);
+        return rootProjectPackageJson.dependencyPackage("react-native").version()
```

Comment:
As I can see this check is the same like in `getReactNativePackageVersionFromNodeModules`. Could you please help me to understand why?

Reply:
The function `getReactNativeVersionFromProjectPackage(cwd: string)` checks if there is React Native version in a project's package.json. But the function `getReactNativePackageVersionFromNodeModules(reactNativePackageDir: string)` checks if there is React Native version in `node_modules/react-native/package.json`.